book_source/source/2darrays.md

```diff
@@ -0,0 +1,237 @@
+# <i class="fas fa-book fa-fw"></i> 2DArrays
+
+## 2DArray Fundamentals
+
+An 2D array is very similar to an 1D array, or just an array as we have learned, only that instead of storing primiative types, it stores another array. In other words, a 2D array is an array where each element contains another array. Commonly this is represented/used when dealing with images, graphs, tables, and various other 2D representations.
+
+Much like how you would access a specific element in an array using square brackets '[]', for a 2D array, you need to add one more pair of brackets to access an element in a 2D array. For example, if you wanted to create a 2D array storing integers, an example syntax could be int[][] (read as a 2D array of ints or 2D int array). Although the specific lengths of the arrays can vary, these arrays still have a fixed length once created. For ease of representation, we typically repersent a 2D array with rows being the inner arrays and columns being the array of arrays. For example:
+```
+[[0 , 1 , 2 , 3 , 4],
+[5 , 6 , 7 , 8 , 9 ],
+[10, 11, 12, 13, 14],
+[15, 16, 17, 18, 19],
+[20, 21, 22, 23, 24]]
+```
+Notice there is an extra pair of opening and closing brackets, which represents the array that stores all of the other arrays. The syntax is commonly represented as array[row][col] and is represented visually as so above for clarity. Additionally, the call to an array using the syntax array[1] would still function, only instead of providing a primative type, it would instead provide the array stored at that index of the array of arrays.
```

Comment:
I think we can drop this paragraph -- the same general information has been covered already.

Reply:
I think it's fine to reinforce this here.